app/src/main/java/org/mozilla/focus/widget/DpadKeyMovementMethod.kt

```diff
@@ -0,0 +1,17 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.widget
+
+import android.text.Spannable
+import android.text.method.ArrowKeyMovementMethod
+import android.widget.TextView
+
+/**
+ * An ArrowKeyMovementMethod that eliminates end cursor selection onFocus
+ */
+class DpadKeyMovementMethod : ArrowKeyMovementMethod() {
```

Comment:
Nit: Suggest changing the class name to more closely describe what it's doing.  Maybe something like `IgnoreFocusMovementMethod`?  I feel that would make the code easier to read from `BrowserNavigationOverlay`.

Reply:
Sounds reasonable. Just for food for thought though - cross referencing ArrowKeyMovementMethod suggests that it is designed to accommodate onTouchEvent more so than onKeyEvent. The reason I named it DpadKeyMovementMethod is so that we have more controls over manipulating movement methods specific to remote control needs moving forward. What do you think is more ideal naming convention for classes (vs. methods) - immediate needs or scalability?

Reply:
Personally, I don't see it as a scalability issue.  If we update the class to handle other aspects of movement we can change the name to match.  But right now the only use of this is to fix this bug.  IMO, if we can make it clear what's happening at the top level, so when reading this I don't have to drill down to find out what's going on, that's a win for readability.